perf/src/sigverify.rs

```diff
@@ -608,15 +610,38 @@ pub fn shrink_batches(batches: &mut Vec<PacketBatch>) {
 
 pub fn ed25519_verify_cpu(batches: &mut [PacketBatch], reject_non_vote: bool, packet_count: usize) {
     debug!("CPU ECDSA for {}", packet_count);
-    PAR_THREAD_POOL.install(|| {
-        batches.into_par_iter().for_each(|batch| {
-            batch.par_iter_mut().for_each(|packet| {
-                if !packet.meta.discard() && !verify_packet(packet, reject_non_vote) {
-                    packet.meta.set_discard(true);
-                }
-            })
+
+    let desired_thread_count = packet_count
+        .saturating_add(VERIFY_MIN_PACKETS_PER_THREAD)
+        .saturating_div(VERIFY_MIN_PACKETS_PER_THREAD);
+    if desired_thread_count >= get_thread_count() {
+        // When using all available threads, skip the overhead of flattening, collecting, etc.
+        PAR_THREAD_POOL.install(|| {
+            batches.into_par_iter().for_each(|batch| {
+                batch.par_iter_mut().for_each(|packet| {
+                    if !packet.meta.discard() && !verify_packet(packet, reject_non_vote) {
+                        packet.meta.set_discard(true);
+                    }
+                })
+            });
         });
-    });
+    } else {
+        let packets_per_thread = packet_count.saturating_div(desired_thread_count);
```

Comment:
`desired_thread_count` won't be zero?

Reply:
1. This could only happen if packet count was somehow negative, which should be guaranteed to never happen
2. I changed the code to not use rayon when we only want to spin up a single thread, which should protect us from this possibility altogether